libs/ast_generic/AST_generic.ml

```diff
@@ -2405,3 +2395,48 @@ let is_metavar_name s = Common.( =~ ) s "^\\(\\$[A-Z_][A-Z_0-9]*\\)$"
 (* coupling: Metavariable.is_metavar_ellipsis *)
 let is_metavar_ellipsis s =
   Common.( =~ ) s "^\\(\\$\\.\\.\\.[A-Z_][A-Z_0-9]*\\)$"
+
+(*****************************************************************************)
+(* Custom visitors *)
+(*****************************************************************************)
+
+(* Most clients should use this instead of the default `iter`. In many cases,
+ * it's not desirable to recurse into id_info since it contains resolved names
+ * and svalues which often contain nodes that are already present elsewhere in
+ * the AST. This matches the default behavior of the old mk_visitor. *)
+class virtual ['self] iter_no_id_info =
+  object (_self : 'self)
+    inherit ['self] iter
+    method! visit_id_info _env _info = ()
+  end
+
+(* This is based on the old behavior of Map_AST.mk_visitor. If you are adding a
```

Comment:
To someone new "the old behavior of Map_AST.mk_visitor" wouldn't mean much. I would find it more helpful to say "This is tailored to Map_AST, see comment in Map_AST about why it makes certain choices" (if such a comment exists).

In contrast the comment above is great because it explains what's special about iter_no_id_info

Reply:
I will elaborate a bit, but I'm going to delete most, if not all of `Map_AST` shortly so referring to a comment there won't help much.